target/linux/layerscape/image/Makefile

```diff
@@ -0,0 +1,70 @@
+#
+# Copyright (C) 2016 Jiang Yutang <yutang.jiang@nxp.com>
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+include $(TOPDIR)/rules.mk
+include $(INCLUDE_DIR)/image.mk
+
+define Build/mk_firmware
+	$(call Image/BuildDTB,$(DTS_DIR)/$(word 1,$(1)).dts,$(DTS_DIR)/$(word 1,$(1)).dtb)
+	dd if=/dev/zero bs=$(word 2,$(1)) count=1 | sed 's/\x00/\xff/g' > $(DTS_DIR)/$(word 1,$(1)).dtb.par
+	dd if=$(DTS_DIR)/$(word 1,$(1)).dtb of=$(DTS_DIR)/$(word 1,$(1)).dtb.par conv=notrunc
+
+	dd if=/dev/zero bs=$(word 3,$(1)) count=1 | sed 's/\x00/\xff/g' > $(KDIR)/$(word 5,$(1))$(KERNEL_SUFFIX).par
+	dd if=$(KDIR)/$(word 5,$(1))$(KERNEL_SUFFIX) of=$(KDIR)/$(word 5,$(1))$(KERNEL_SUFFIX).par conv=notrunc
+
+	dd if=/dev/zero bs=$(word 4,$(1)) count=1 | sed 's/\x00/\xff/g' > $(KDIR)/root.squashfs.$(word 5,$(1)).par
+	dd if=$(KDIR)/root.squashfs of=$(KDIR)/root.squashfs.$(word 5,$(1)).par conv=notrunc
+
+	cat $(DTS_DIR)/$(word 1,$(1)).dtb.par $(KDIR)/$(word 5,$(1))$(KERNEL_SUFFIX).par $(KDIR)/root.squashfs.$(word 5,$(1)).par > $@
+endef
```

Comment:
mk_firmware is doing too much stuff at once, also it should never refer to something like root.squashfs directly.
Does the padding really need to be with 0xff? If not, you could use the generic pad-to template.
If you need 0xff, then maybe you should make a build template that adds a file and pads it.
To get the rootfs filename, you can refer to $(IMAGE_ROOTFS)